src/heuristics.rs

```diff
@@ -156,7 +156,13 @@ impl HeuristicTests {
         log::trace!("enter: make_wildcard_request({}, {})", target, length);
 
         let unique_str = self.unique_string(length);
-        let nonexistent_url = target.format(&unique_str, None)?;
+        // To take care of slash when needed
```

Comment:
I agree with you that does not change alot but some server are configured in such a way that they don't try to add trailing slash in url wich mean that the server doesn't recognize /cgi-bin but /cgi-bin/ yes, (example with a machine in OSCP).

You can see a stackoverflow response [here](https://stackoverflow.com/questions/1188927/why-the-trailing-slash-in-the-web-service-is-so-important) about trailing slash.

"Many webservers will automatically correct such URLs, but it is not required for them to do so."
Example:
![cgi_bin_with_slash](https://user-images.githubusercontent.com/34527333/134447386-eec8431a-2204-45d0-a20d-c7b1035b4a88.png)
![cgi_bin_without_slash](https://user-images.githubusercontent.com/34527333/134447387-0247c628-f122-43d5-aba1-556a77834ec9.png)

That's my take on it, in fact it might be even smarter to check for both with and without trailing slash Or letting the user deal with it by using the add slash configuration. 
If he knows the server need it (the reason why i wanted to implement it with extension in the first place) he can add that option and the test will be performed with an ending slash in that configuration ?